kubetest2/kubetest2-gce/deployer/up.go

```diff
@@ -0,0 +1,96 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package deployer
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"k8s.io/test-infra/kubetest2/pkg/exec"
+)
+
+func (d *deployer) Up() error {
+	if err := d.verifyFlags(); err != nil {
+		return fmt.Errorf("up could not verify flags: %s", err)
+	}
+
+	if err := d.enableComputeAPI(); err != nil {
+		return fmt.Errorf("up couldn't enable compute API: %s", err)
+	}
+
+	env := d.buildEnv()
+	script := filepath.Join(d.RepoRoot, "cluster", "kube-up.sh")
+
+	cmd := exec.Command(script)
+	cmd.SetEnv(env...)
+	exec.InheritOutput(cmd)
+	err := cmd.Run()
+	if err != nil {
+		return fmt.Errorf("error encountered during %s: %s", script, err)
+	}
+
+	return nil
+}
+
+func (d *deployer) buildEnv() []string {
+	env := []string{}
+
+	// path is necessary for scripts to find gsutil, gcloud, etc
+	env = append(env, fmt.Sprintf("PATH=%s", os.Getenv("PATH")))
+	env = append(env, fmt.Sprintf("PROJECT=%s", d.GCPProject))
+	env = append(env, fmt.Sprintf("KUBECONFIG=%s", d.kubeconfigPath))
```

Comment:
adding to the previous comment e.g. IIUC KUBECONFIG here is set so that kube-up will populate it for us right? and not us providing it like what kubetest2 core will do for the tester eventually.

Reply:
Clarifying this in a comment. Yes, this makes kube-up create the kubeconfig at a specified path (in the artifacts directory) so that kubetest2 core can eventually ask the deployer "where is the kubeconfig" and it can always respond with the same path (in the artifacts directory).